internal/ui/format.go

```diff
@@ -105,3 +107,24 @@ func ToJSONString(status interface{}) string {
 	}
 	return buf.String()
 }
+
+// TerminalDisplayWidth returns the number of terminal cells needed to display s
+func TerminalDisplayWidth(s string) int {
+	width := 0
+	for _, r := range s {
+		width += terminalDisplayRuneWidth(r)
+	}
+
+	return width
+}
+
+func terminalDisplayRuneWidth(r rune) int {
```

Comment:
I've tried to unify this with `wideRune` in `internal/ui/termstatus/status.go`. However, it turned out that the use cases are somewhat different. In status, it is better to overestimate the rune width if in doubt. So let's keep separate functions for now.